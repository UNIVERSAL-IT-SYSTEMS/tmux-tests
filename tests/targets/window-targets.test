#!/bin/sh

$TMUX_L new -d -sS -nzero -x 80 -y 25 "$TMUX_L wait never"
$TMUX_L set -g default-command "$TMUX_L wait never"

$TMUX_L splitw
for i in one middle three four; do
	$TMUX_L neww -n$i\; splitw
done

try_target1()
{
	$TMUX_L selectw -t$1 2>&1
	result=$(echo $($TMUX_L display -pF'#{window_name},#{window_id},#{window_index}.#{pane_index}'))
	printf "%-24s %s\n" "$1" "$result"
}

try_target()
{
	eval "$2"; try_target1 "$1"
	eval "$2"; try_target1 "$1."
	eval "$2"; try_target1 "$1.0"
	eval "$2"; try_target1 "$1.1"
	eval "$2"; try_target1 ":$1"
	eval "$2"; try_target1 ":$1."
	eval "$2"; try_target1 ":$1"
	eval "$2"; try_target1 ":$1.0"
	eval "$2"; try_target1 ":$1.1"
	eval "$2"; try_target1 "S:$1"
	eval "$2"; try_target1 "S:$1."
	eval "$2"; try_target1 "S:$1.0"
	eval "$2"; try_target1 "S:$1.1"
}

try_target "{last}" "$TMUX_L selectw -t:one\; selectp -t0\; selectw -t:middle\; selectp -t0"
try_target "!" "$TMUX_L selectw -t:one\; selectp -t0\; selectw -t:middle\; selectp -t0"

read targets <<EOF
{start} \
^ \
{end} \
\$ \
{next} \
+ \
{previous} \
- \
one \
middle \
three \
four \
0 \
1 \
2 \
3 \
4 \
{}
EOF
for target in $targets; do
	try_target "$target" "$TMUX_L selectw -t:middle\; selectp -t1"
done
