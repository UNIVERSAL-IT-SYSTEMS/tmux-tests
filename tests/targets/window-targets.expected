{last}                   one,@1,1.0
{last}.                  one,@1,1.0
{last}.0                 one,@1,1.0
{last}.1                 one,@1,1.0
:{last}                  one,@1,1.0
:{last}.                 one,@1,1.0
:{last}                  one,@1,1.0
:{last}.0                one,@1,1.0
:{last}.1                one,@1,1.0
S:{last}                 one,@1,1.0
S:{last}.                one,@1,1.0
S:{last}.0               one,@1,1.0
S:{last}.1               one,@1,1.0
!                        one,@1,1.0
!.                       one,@1,1.0
!.0                      one,@1,1.0
!.1                      one,@1,1.0
:!                       one,@1,1.0
:!.                      one,@1,1.0
:!                       one,@1,1.0
:!.0                     one,@1,1.0
:!.1                     one,@1,1.0
S:!                      one,@1,1.0
S:!.                     one,@1,1.0
S:!.0                    one,@1,1.0
S:!.1                    one,@1,1.0
{start}                  zero,@0,0.1
{start}.                 zero,@0,0.1
{start}.0                zero,@0,0.1
{start}.1                zero,@0,0.1
:{start}                 zero,@0,0.1
:{start}.                zero,@0,0.1
:{start}                 zero,@0,0.1
:{start}.0               zero,@0,0.1
:{start}.1               zero,@0,0.1
S:{start}                zero,@0,0.1
S:{start}.               zero,@0,0.1
S:{start}.0              zero,@0,0.1
S:{start}.1              zero,@0,0.1
^                        zero,@0,0.1
^.                       zero,@0,0.1
^.0                      zero,@0,0.1
^.1                      zero,@0,0.1
:^                       zero,@0,0.1
:^.                      zero,@0,0.1
:^                       zero,@0,0.1
:^.0                     zero,@0,0.1
:^.1                     zero,@0,0.1
S:^                      zero,@0,0.1
S:^.                     zero,@0,0.1
S:^.0                    zero,@0,0.1
S:^.1                    zero,@0,0.1
{end}                    four,@4,4.1
{end}.                   four,@4,4.1
{end}.0                  four,@4,4.1
{end}.1                  four,@4,4.1
:{end}                   four,@4,4.1
:{end}.                  four,@4,4.1
:{end}                   four,@4,4.1
:{end}.0                 four,@4,4.1
:{end}.1                 four,@4,4.1
S:{end}                  four,@4,4.1
S:{end}.                 four,@4,4.1
S:{end}.0                four,@4,4.1
S:{end}.1                four,@4,4.1
can't find session $
$                        middle,@2,2.1
$.                       four,@4,4.1
$.0                      four,@4,4.1
$.1                      four,@4,4.1
:$                       four,@4,4.1
:$.                      four,@4,4.1
:$                       four,@4,4.1
:$.0                     four,@4,4.1
:$.1                     four,@4,4.1
S:$                      four,@4,4.1
S:$.                     four,@4,4.1
S:$.0                    four,@4,4.1
S:$.1                    four,@4,4.1
{next}                   three,@3,3.1
{next}.                  three,@3,3.1
{next}.0                 three,@3,3.1
{next}.1                 three,@3,3.1
:{next}                  three,@3,3.1
:{next}.                 three,@3,3.1
:{next}                  three,@3,3.1
:{next}.0                three,@3,3.1
:{next}.1                three,@3,3.1
S:{next}                 three,@3,3.1
S:{next}.                three,@3,3.1
S:{next}.0               three,@3,3.1
S:{next}.1               three,@3,3.1
+                        three,@3,3.1
+.                       three,@3,3.1
+.0                      three,@3,3.1
+.1                      three,@3,3.1
:+                       three,@3,3.1
:+.                      three,@3,3.1
:+                       three,@3,3.1
:+.0                     three,@3,3.1
:+.1                     three,@3,3.1
S:+                      three,@3,3.1
S:+.                     three,@3,3.1
S:+.0                    three,@3,3.1
S:+.1                    three,@3,3.1
{previous}               one,@1,1.0
{previous}.              one,@1,1.0
{previous}.0             one,@1,1.0
{previous}.1             one,@1,1.0
:{previous}              one,@1,1.0
:{previous}.             one,@1,1.0
:{previous}              one,@1,1.0
:{previous}.0            one,@1,1.0
:{previous}.1            one,@1,1.0
S:{previous}             one,@1,1.0
S:{previous}.            one,@1,1.0
S:{previous}.0           one,@1,1.0
S:{previous}.1           one,@1,1.0
-                        one,@1,1.0
-.                       one,@1,1.0
-.0                      one,@1,1.0
-.1                      one,@1,1.0
:-                       one,@1,1.0
:-.                      one,@1,1.0
:-                       one,@1,1.0
:-.0                     one,@1,1.0
:-.1                     one,@1,1.0
S:-                      one,@1,1.0
S:-.                     one,@1,1.0
S:-.0                    one,@1,1.0
S:-.1                    one,@1,1.0
one                      one,@1,1.0
one.                     one,@1,1.0
one.0                    one,@1,1.0
one.1                    one,@1,1.0
:one                     one,@1,1.0
:one.                    one,@1,1.0
:one                     one,@1,1.0
:one.0                   one,@1,1.0
:one.1                   one,@1,1.0
S:one                    one,@1,1.0
S:one.                   one,@1,1.0
S:one.0                  one,@1,1.0
S:one.1                  one,@1,1.0
middle                   middle,@2,2.1
middle.                  middle,@2,2.1
middle.0                 middle,@2,2.1
middle.1                 middle,@2,2.1
:middle                  middle,@2,2.1
:middle.                 middle,@2,2.1
:middle                  middle,@2,2.1
:middle.0                middle,@2,2.1
:middle.1                middle,@2,2.1
S:middle                 middle,@2,2.1
S:middle.                middle,@2,2.1
S:middle.0               middle,@2,2.1
S:middle.1               middle,@2,2.1
three                    three,@3,3.1
three.                   three,@3,3.1
three.0                  three,@3,3.1
three.1                  three,@3,3.1
:three                   three,@3,3.1
:three.                  three,@3,3.1
:three                   three,@3,3.1
:three.0                 three,@3,3.1
:three.1                 three,@3,3.1
S:three                  three,@3,3.1
S:three.                 three,@3,3.1
S:three.0                three,@3,3.1
S:three.1                three,@3,3.1
four                     four,@4,4.1
four.                    four,@4,4.1
four.0                   four,@4,4.1
four.1                   four,@4,4.1
:four                    four,@4,4.1
:four.                   four,@4,4.1
:four                    four,@4,4.1
:four.0                  four,@4,4.1
:four.1                  four,@4,4.1
S:four                   four,@4,4.1
S:four.                  four,@4,4.1
S:four.0                 four,@4,4.1
S:four.1                 four,@4,4.1
0                        zero,@0,0.1
0.                       zero,@0,0.1
0.0                      zero,@0,0.1
0.1                      zero,@0,0.1
:0                       zero,@0,0.1
:0.                      zero,@0,0.1
:0                       zero,@0,0.1
:0.0                     zero,@0,0.1
:0.1                     zero,@0,0.1
S:0                      zero,@0,0.1
S:0.                     zero,@0,0.1
S:0.0                    zero,@0,0.1
S:0.1                    zero,@0,0.1
1                        one,@1,1.0
1.                       one,@1,1.0
1.0                      one,@1,1.0
1.1                      one,@1,1.0
:1                       one,@1,1.0
:1.                      one,@1,1.0
:1                       one,@1,1.0
:1.0                     one,@1,1.0
:1.1                     one,@1,1.0
S:1                      one,@1,1.0
S:1.                     one,@1,1.0
S:1.0                    one,@1,1.0
S:1.1                    one,@1,1.0
2                        middle,@2,2.1
2.                       middle,@2,2.1
2.0                      middle,@2,2.1
2.1                      middle,@2,2.1
:2                       middle,@2,2.1
:2.                      middle,@2,2.1
:2                       middle,@2,2.1
:2.0                     middle,@2,2.1
:2.1                     middle,@2,2.1
S:2                      middle,@2,2.1
S:2.                     middle,@2,2.1
S:2.0                    middle,@2,2.1
S:2.1                    middle,@2,2.1
3                        three,@3,3.1
3.                       three,@3,3.1
3.0                      three,@3,3.1
3.1                      three,@3,3.1
:3                       three,@3,3.1
:3.                      three,@3,3.1
:3                       three,@3,3.1
:3.0                     three,@3,3.1
:3.1                     three,@3,3.1
S:3                      three,@3,3.1
S:3.                     three,@3,3.1
S:3.0                    three,@3,3.1
S:3.1                    three,@3,3.1
4                        four,@4,4.1
4.                       four,@4,4.1
4.0                      four,@4,4.1
4.1                      four,@4,4.1
:4                       four,@4,4.1
:4.                      four,@4,4.1
:4                       four,@4,4.1
:4.0                     four,@4,4.1
:4.1                     four,@4,4.1
S:4                      four,@4,4.1
S:4.                     four,@4,4.1
S:4.0                    four,@4,4.1
S:4.1                    four,@4,4.1
can't find window {}
{}                       middle,@2,2.1
can't find window {}
{}.                      middle,@2,2.1
can't find window {}
{}.0                     middle,@2,2.1
can't find window {}
{}.1                     middle,@2,2.1
can't find window {}
:{}                      middle,@2,2.1
can't find window {}
:{}.                     middle,@2,2.1
can't find window {}
:{}                      middle,@2,2.1
can't find window {}
:{}.0                    middle,@2,2.1
can't find window {}
:{}.1                    middle,@2,2.1
can't find window {}
S:{}                     middle,@2,2.1
can't find window {}
S:{}.                    middle,@2,2.1
can't find window {}
S:{}.0                   middle,@2,2.1
can't find window {}
S:{}.1                   middle,@2,2.1
